### ISSUE TITLE
Empty object: "This type may be different from what you expect it to be"

### ISSUE THREAD

**Issue by oganexon** [denoland/deno_lint#774]
### Lint Name

`ban-types`

### Code Snippet

```ts
type User = {
  id: string;
  name: string;
};

type ApiResponse = {
  nodes: (User | {})[];
};
```

### Expected Result

Linter should not complain because I expect `{}` to be an empty object. Perhaps there's another way to type this?

### Actual Result

```
❯ deno lint
(ban-types) This type may be different from what you expect it to be
  nodes: (User | {})[];
                 ^^
    at [file]

    hint: If you want a type meaning "any object", use `Record<string, unknown>` instead. Or if you want a type meaning "any value", you probably want `unknown` instead.
Found 1 problem
Checked 1 file
```

### Additional Info

I use an API that returns an array of `User` objects and empty objects. This example is way simplified, but this is the idea.

### Version

deno 1.11.3 (release, x86_64-pc-windows-msvc)
v8 9.1.269.35
typescript 4.3.2

**Comment by magurotuna** [comment]
@oganexon 
If you want to specify an empty object type, try using `Record<string, never>` instead of `{}`. 

```ts
type ApiResponse = {
  nodes: (User | Record<string, never>)[];
}
```

Alternatively the following also works:

```ts
type ApiResponse = {
  nodes: (User | {[k: string]: never})[];
}
```

The linter should add this information to the hint. I'll do it.

**Comment by oganexon** [comment]
Thanks, I didn't think of that!
